src/lowering/desugar.ml

```diff
@@ -1082,13 +1145,31 @@ let import_unit (u : S.comp_unit) : import_declaration =
       let class_typ = match List.map T.normalize ts2 with
         | [ T.Async(_, t2) ] -> t2
         | _ -> assert false in
-      let func = funcE id T.Local T.Returns
-        [typ_arg c T.Scope T.scope_bound]
-        as_
-        [T.Async (List.hd cs, actor_t)]
-        body
+      let install_arg =
+        fresh_var "install_arg" T.install_arg_typ
+      in
+      let installBody =
+        funcE id T.Local T.Returns
+          [typ_arg c T.Scope T.scope_bound]
+          as_
+          [T.Async (List.hd cs, actor_t)]
+          body
+      in
+      let install =
+        funcE id T.Local T.Returns
+          []
+          ([arg_of_var install_arg])
+          [installBody.note.Note.typ]
+          (ifE
+             (primE (Ir.RelPrim (T.install_arg_typ, Operator.EqOp))
+               [ varE install_arg;
+                 tagE "new" (recordE ["settings", nullE()]) ])
+             installBody
+             (primE (Ir.OtherPrim "trap")
```

Comment:
Do we compile this into the Wasm too? Isn't there an IR primitive that prunes interpreter control-flow?

Reply:
No, this only used in the interpreter, never compiled.